### DISCUSSION TITLE
Fortran error from xmltools.f90

### DISCUSSION THREAD

**Discussion by harishpvv** [anharmonic/d3q#1] [Q&A] [answered]
Sir, I installed d3q.x, and when I tried to run it is showing the following error. For installing, I did 'make d3q' in the root qe(v 6.7) folder.

Error termination. Backtrace:
At line 965 of file xmltools.f90
Fortran runtime error: Sequential READ or WRITE not allowed after EOF marker, possibly use REWIND or BACKSPACE

can you please check this.
Thanks.

Harish

**Comment by paulatz** [comment, marked as answer]
Yes, there is a little bug in the restart mechanism. You can work around it by starting each triple separately (by using first= and last= input variables), which is what I always do to compute them simultaneously. Or you can replace d3matrix_io2.f90 with this one
https://raw.githubusercontent.com/anharmonic/d3q/main/src/d3matrix_io2.f90 ***@***.***/0?redirect=https%3A%2F%2Fraw.githubusercontent.com%2Fanharmonic%2Fd3q%2Fmain%2Fsrc%2Fd3matrix_io2.f90&recipient=cmVwbHkrQUFKVlBKRUdaVTQ0TlJCUlhSMk5SQk42TDU0M1pFVkJNWEhBQU1RTjI0QHJlcGx5LmdpdGh1Yi5jb20%3D)
and recompile.
sorry for the inconvenience
--
Lorenzo Paulatto - Paris
